code/modules/clothing/head/_head.dm

```diff
@@ -7,6 +7,8 @@
 	slot_flags = ITEM_SLOT_HEAD
 	var/blockTracking = 0 //For AI tracking
 	var/can_toggle = null
+	var/snug_fit = FALSE //is the hat immune to being knocked off?
+	var/anti_tinfoil_maneuver = FALSE //if the hat does something negative when worn, this should probably be set to TRUE

```

Comment:
what's the difference between these two vars?

Rather, why not just give the tinfoil hat `snug_fit` instead of giving every headpiece a special var?

Reply:
snug_fit checks if the hat being worn can be knocked off by incoming hats

the tinfoil one prevents the hat from ever potentially landing on someone's head

Reply:
everywhere the anti_tinfoil check is used there's also a snug_fit check, so i'm still not seeing why you wouldn't just use snug fit

Reply:
when hat gets thrown and the person hit by the hat is already wearing a hat, snug_fit is if the worn hat can get knocked off and replaced. it exists mostly for hardsuit helmets and wizard hats, but also prevents security helmets and hardhats from getting replaced.

the tinfoil check is specifically for preventing a hat from landing on another person's head.

Reply:
I see now